### PR TITLE
Hide configurations that are handled by Cloud

### PR DESCRIPTION
**Pull request by ranquild**
**Context**
Some users want to tweak settings that are handled by us which end up on a ticket or on an error. 

**What**
Let's remove them, so users can't shoot themselves in the foot. They are:
- Admin → Settings → General → Site URL
- Admin → Settings → General → Redirect to HTTPS
- Admin → Settings → Email (remove the panel from the sidebar altogether)
- Admin → Settings → Updates
- Admin → Enterprise

### CODE REVIEW

File: frontend/test/metabase/scenarios/admin/settings/settings.cy.spec.js [hunk 1]
```diff
@@ -356,6 +356,17 @@ describe("scenarios > admin > settings", () => {
...
+    setupMetabaseCloud();
+    cy.visit("/admin/settings/general");
+
+    cy.findByText("Site Name");
+    cy.findByText("Site URL").should("not.exist");
+
+    cy.findByText("Email").should("not.exist");
+    cy.findByText("Updates").should("not.exist");
+  });
+
```
Comment (kulyk) [inline frontend/test/metabase/scenarios/admin/settings/settings.cy.spec.js:369]: You can also use `describeWithToken` helper from `__support__/e2e/cypress` to test the Enterprise tab is also hidden. See an example [here](https://github.com/metabase/metabase/blob/master/frontend/test/metabase/scenarios/admin/audit/ad-hoc.cy.spec.js)

**Review by kulyk — APPROVED** [review]
Looks good, just have a suggestion on testing if the "Enterprise" tab gets hidden

File: frontend/test/metabase/scenarios/admin/settings/settings.cy.spec.js (hunk 1 above)
Comment (ranquild) [reply frontend/test/metabase/scenarios/admin/settings/settings.cy.spec.js:370]: Great, thanks!

**Comment by ranquild** [comment]
@metabase-bot run visual tests

**Pull request merged**